core/src/main/java/org/neo4j/ogm/metadata/ClassInfo.java

```diff
@@ -1098,16 +1111,28 @@ public FieldInfo getEndNodeReader() {
      * @return a FieldInfo for the field annotated as the StartNode, or none if not found
      */
     public FieldInfo getStartNodeReader() {
+        if (startNodeReader == null) {
+            initStartNodeReader();
+        }
+        return startNodeReader.orElse(null);
```

Comment:
```suggestion
        return startNodeReader.orElseGet(() -> {
                 LOGGER.warn("Failed to find an @StartNode on {}", name());
                 return null;
        });
```

Reply:
I did not do that because it somehow change the logic. `null` is also the value when `isRelationshipEntity()` is false. It will mean that the warning will be outputed as well in that case. Is it ok for you?

Reply:
Aaah, good point!

Reply:
Your solution is better!